homeassistant/components/homewizard/coordinator.py

```diff
@@ -74,7 +74,7 @@ async def _async_update_data(self) -> DeviceResponseEntry:
 
                 # Do not reload when performing first refresh
                 if self.data is not None:
-                    await self.hass.config_entries.async_reload(
+                    self.hass.config_entries.async_schedule_reload(
                         self.config_entry.entry_id
                     )
```

Comment:
Having the tests makes me think...
Why not trigger directly an authentication error?

It feels like you are reloading the integration when you know the authentication is bad

Reply:
Not really sure if this still is the best solution, but it was chosen to use a reload via this review: https://github.com/home-assistant/core/pull/82366#discussion_r1032810952

Reply:
Sure!